src/navigationGuards.ts

```diff
@@ -253,6 +253,18 @@ export function extractComponentsGuards(
           )
           let promise = rawComponent
           rawComponent = () => promise
+        } else if (
+          (rawComponent as any).__asyncLoader &&
+          !(rawComponent as any).__resolvedAsyncComponent
```

Comment:
let's remove the `__resolvedAsyncComponent` altogether and let the user fix the problem by themselves